### PR TITLE
NAV-229-new-skip-logic

### PR DESCRIPTION
**Pull request by Manoj-nathwani**
https://fjelltopp.atlassian.net/browse/NAV-229

# Tests
```
$ yarn test lib/actionButtons.test.js

 PASS  lib/actionButtons.test.js
  The "Task Complete" button should
    if the workflow is !reached
      ✓ always return a disabled button with no onClick Action (3 ms)
    if the workflow is terminus
      ✓ always return a disabled button with no onClick Action
    if the workflow is reached and !terminus
      if the task isLatestTask in the breadcrumbs
        and is set to manual
          ✓ and is incomplete, should return an enabled unchecked button with the markTaskAsComplete onclick action (1 ms)
        and is set to not manual
          ✓ and is incomplete, should return an enabled unchecked button with the toggleCompleteStateLocally onclick action (1 ms)
      if the task !isLatestTask in the breadcrumbs
        and is set to manual
          ✓ and is incomplete, should return an enabled unchecked button with the markTaskAsComplete onclick action
          ✓ and is complete, should return an enabled checked button with the markTaskAsIncomplete onclick action
        and is set to not manual
          ✓ and is incomplete, should return an disabled unchecked button
          ✓ and is complete, should return an disabled checked button
  The "Prior Task" button should
    if the workflow is !reached
      ✓ the "Prior Task" button should be disabled (1 ms)
    if the task is reached
      if the workflow is terminus
        ✓ always return a disabled button with no onClick Action
      if the workflow is !terminus
        ✓ if isOldestTask in the breadcrumbs, should return a disabled button
        if !isOldestTask in the breadcrumbs
          ✓ and if isLatestTask in the breadcrumbs, should return a button with getPreviousTask onclick action
          and if !isLatestTask in the breadcrumbs
            ✓ and if task is complete should return a button with onclick getPreviousTask action (6 ms)
            ✓ and if task is incomplete should return a button with onclick skipTaskAndGetPreviousTask action
  The "Next Task" button should
    if the workflow is !reached
      ✓ the "Next Task" button should be disabled
    if the task is reached
      ✓ if the workflow is terminus, should return a disabled button
      if the workflow is !terminus
        ✓ and if isLatestTask in the breadcrumbs, should return a disabled button (1 ms)
        and if !isLatestTask in the breadcrumbs
          ✓ and if the task is complete, should return a button with onclick getNextTask action (1 ms)
          ✓ and if the task is incomplete, should return a button with onclick skipTaskAndGetNextTask action
  The "What's Next" button should
    ✓ if the task is complete, should return a button with onclick fetchLatestWorkflowState actions (1 ms)
    ✓ if the task is incomplete, should return a button with onclick skipTaskAndFetchLatestWorkflowState actions (1 ms)

Test Suites: 1 passed, 1 total
Tests:       21 passed, 21 total
Snapshots:   0 total
Time:        2.049 s
Ran all test suites matching /lib\/actionButtons.test.js/i.
Done in 3.51s.
```

### CODE REVIEW

File: lib/actionButtons.test.js [hunk 1]
```diff
@@ -104,78 +104,77 @@ describe('The "Task Complete" button should', () => {
...
-            const taskBreadcrumbs = ['task1', 'task2'];
-            const workflowState = { currentTask: { id, manual, details }, taskBreadcrumbs };
-            expect(priorTaskButton(workflowState))
-                .toMatchObject({ enabled: false, onClickAction: null });
-        });
-    });
-    describe('If the task is not set to manual', () => {
-        const manual = false;
-        test('and if breadcrumbs length > 1, should return an enabled button with the getPreviousTask onclick action', () => {
+    const manual = true; // TODO: confirm we no longer care if a task is manual or not
```
Comment (Manoj-nathwani) [inline lib/actionButtons.test.js:107]: @jonathansberry can you confirm this please

Comment (jonathansberry) [reply lib/actionButtons.test.js:107]: For the prior task button, no we do not care if it is manual.

**Review by jonathansberry — CHANGES_REQUESTED** [review]
First thing hits me is we have lost the symmetry between the next task and prior task buttons (there's an extra if for next task) which immediately sets of alarm bells that something is wrong.  Basically I think prior task should be disabled when viewing an unreached task (like next task).

**Comment by Manoj-nathwani** [comment]
> First thing hits me is we have lost the symmetry between the next task and prior task buttons (there's an extra if for next task) which immediately sets of alarm bells that something is wrong. Basically I think prior task should be disabled when viewing an unreached task (like next task).

image

This has now been addressed ✅

**Review by jonathansberry — APPROVED** [review]

**Pull request merged**